df_auth/drf/serializers.py

```diff
@@ -125,18 +122,30 @@ class FirstLastNameSerializerMixin(serializers.Serializer):
     last_name = serializers.CharField(required=False)
 
 
-class SocialTokenObtainSerializer(FirstLastNameSerializerMixin, TokenCreateSerializer):
-    access_token = serializers.CharField(write_only=True)
-    provider = serializers.ChoiceField(
-        choices=[
-            (backend.name, backend.name)
-            for backend in AUTHENTICATION_BACKENDS
-            if hasattr(backend, "name")
-        ],
-    )
-
+class SocialTokenObtainBaseSerializer(
+    FirstLastNameSerializerMixin, TokenCreateSerializer
+):
     response = serializers.JSONField(read_only=True)
 
+    def validate(self, attrs):
```

Comment:
You have code duplication here. The only difference between OAuth1 and OAuth2 is 

```
self.user = request.backend.do_auth(attrs["access_token"], user=user)
```

and 

```
self.user = request.backend.do_auth({
            "oauth_token": data["oauth_token"],
            "oauth_token_secret": data["oauth_token_secret"],
        }, user=user)
```

Move token data obtaining in a separate method and use it in validate base method